CMakeLists.txt

```diff
@@ -92,31 +92,23 @@ SET(TYPE_ARCHITECTURE "native" CACHE STRING "Choose the type of architecture to
 message(STATUS "Build type is: ${CMAKE_BUILD_TYPE}")
 
 ########## COMPILATION FLAGS ##########
-add_compile_options(-Wall -Wextra)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Weffc++")
-# to avoid conflicts with built-in functions
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-builtin")
-
-
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang"
-    OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
-  # using Clang
-  add_compile_options(-Wno-missing-braces)
-
-elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-  # using GCC
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-non-virtual-dtor")
-  string( TOLOWER "${CMAKE_BUILD_TYPE}" build_type )
-  if ("release" STREQUAL "${build_type}" )
-    add_compile_options("-march=${TYPE_ARCHITECTURE}")
-  endif()
-  if ("debug" STREQUAL "${build_type}" )
-    add_compile_options(-O0)
-  endif()
-elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
-  # using Intel C++
-elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
-  # using Visual Studio C++
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Weffc++")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra")
+
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR
+    "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR
+    "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
+
+    set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -Og")
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Og")
```

Comment:
Do we want to enable this flag by default? As mentioned in #121, this cuts down the runtime of tests to 9s from 147s. The build time goes from 11min (at -O0) to 12min (at -Og).

Reply:
what is the diff between these two optimization flags ? I have a hard time finding it.

Reply:
-O0 is the default and don't optimize anything. See https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html for the options enabled by -Og

Reply:
Seems like `-Og` is meant for this so yes I would use it by default